react/src/utils/fragments.ts

```diff
@@ -0,0 +1,134 @@
+import { gql } from "@apollo/client";
+
+export const PRODUCT_BASIC_FIELDS_FRAGMENT = gql`
+  fragment ProductBasicFields on Product {
+    id
+    name
+    gender
+  }
+`;
+
+export const BOX_FIELDS_FRAGMENT = gql`
+  fragment BoxFields on Box {
+    labelIdentifier
+    state
+    size {
+      id
+      label
+    }
+    numberOfItems
+    comment
+  }
+`;
+
+export const TAG_FIELDS_FRAGMENT = gql`
+  fragment TagFields on Tag {
+    id
+    name
+    color
+  }
+`;
+
+export const TAG_OPTIONS_FRAGMENT = gql`
+  fragment TagOptions on Tag {
+    value: id
+    label: name
+    color
+  }
+`;
+
+export const SIZE_FIELDS_FRAGMENT = gql`
+  fragment SizeFields on Size {
+    id
+    label
+  }
+`;
+
+export const SIZE_RANGE_FIELDS_FRAGMENT = gql`
+  ${SIZE_FIELDS_FRAGMENT}
+  fragment SizeRangeFields on SizeRange {
+    id
+    label
+    sizes {
+      ...SizeFields
+    }
+  }
+`;
+
+export const DISTRO_EVENT_FIELDS_FRAGMENT = gql`
+  fragment DistroEventFields on DistributionEvent {
+    id
+    state
+    name
+    distributionSpot {
+      name
+    }
+    plannedStartDateTime
+    plannedEndDateTime
+  }
+`;
+
+export const PRODUCT_FIELDS_FRAGMENT = gql`
+  ${PRODUCT_BASIC_FIELDS_FRAGMENT}
+  ${SIZE_RANGE_FIELDS_FRAGMENT}
+  fragment ProductFields on Product {
+    ...ProductBasicFields
+    category {
+      name
+    }
+    sizeRange {
+      ...SizeRangeFields
+    }
+  }
+`;
+
+export const BOX_WITH_SIZE_TAG_PRODUCT_FIELDS_FRAGMENT = gql`
+  ${SIZE_FIELDS_FRAGMENT}
+  ${PRODUCT_BASIC_FIELDS_FRAGMENT}
+  ${TAG_FIELDS_FRAGMENT}
+  ${DISTRO_EVENT_FIELDS_FRAGMENT}
+  fragment BoxWithSizeTagProductFields on Box {
+    labelIdentifier
+    state
+    size {
+      ...SizeFields
+    }
+    numberOfItems
+    comment
+    product {
+      ...ProductBasicFields
+    }
+    tags {
+      ...TagFields
+    }
+    distributionEvent {
+      ...DistroEventFields
+    }
+    location {
+      id
+      name
+      ... on ClassicLocation {
+        defaultBoxState
+      }
+      base {
+        locations {
+          id
+          name
+          ... on ClassicLocation {
+            defaultBoxState
+          }
+        }
+        distributionEventsBeforeReturnedFromDistributionState {
```

Comment:
Do you think the distro-event parts can be simply removed (if so, in a follow-up PR but only after the first release I'd say)? Overall it's overhead queries for the BE, and the response data is not used atm in the UI.

Reply:
> Thanks for the quick work @vahidbazzaz ! Using fragments seems to slim down the code a lot, glad you figured it out (though a separate PR would've made the changes easier to distinguish). How did you solve the issue about the graphql-types not being generated?

Thanks @pylipp, it was a little strange. Instead of altering the previous GraphQL query, I created a new one with the `baseId` parameter, then the type generator created new types, and I removed the old one.

Reply:
> Do you think the distro-event parts can be simply removed (if so, in a follow-up PR but only after the first release I'd say)? Overall it's overhead queries for the BE, and the response data is not used atm in the UI.

@pylipp We may need to consult with @aerinsol and @spaudanjo on this, but I don't have any objections.